### PR TITLE
Adding more cookbook examples

### PR DESCRIPTION
**Pull request by miroslav-tashonov**

### CODE REVIEW

**Review by Gusarich — CHANGES_REQUESTED** [review]
@miroslav-tashonov check the review please

File: docs/develop/func/cookbook.md [hunk 1]
```diff
@@ -192,3 +192,293 @@ forall X -> (tuple) to_tuple (X x) asm "NOP";
...
+### Reversing tuples
+
+```
+forall X -> (tuple, X) ~tpop(tuple t) asm "TPOP";
+int tuple_length(tuple t) asm "TLEN";
+forall X -> int cast_to_int(X x) asm "NOP";
+
+(tuple) reverse_tuple(tuple original_tuple)
+{
+    tuple reverted_order_tuple = unsafe_tuple([]);
```
[the remark this answers is not in the document]
Comment (miroslav-tashonov) [reply docs/develop/func/cookbook.md]: Done, ty!

File: docs/develop/func/cookbook.md [hunk 2]
```diff
@@ -192,3 +192,293 @@ forall X -> (tuple) to_tuple (X x) asm "NOP";
...
+    repeat(original_tuple.tuple_length())
+    {
+        var initial = original_tuple~tpop();
+        int result_as_int = cast_to_int(initial);
```
[the remark this answers is not in the document]
Comment (miroslav-tashonov) [reply docs/develop/func/cookbook.md]: Agree, removed it now

File: docs/develop/func/cookbook.md [hunk 3]
```diff
@@ -192,3 +192,293 @@ forall X -> (tuple) to_tuple (X x) asm "NOP";
...
+    return reverted_order_tuple;
+}
+```
+
+### Determine if slices are equal
+
+There are two different ways we can determine the equality. One is based on the slice hash, while the other one by using the SDEQ asm instruction.
+
+```
+int equal_slices(slice a, slice b) asm "SDEQ";
```
[the remark this answers is not in the document]
Comment (miroslav-tashonov) [reply docs/develop/func/cookbook.md:317]: I think it is nice if people are aware of more different ways on how they can implement things, the more different variations the better

File: docs/develop/func/cookbook.md [hunk 4]
```diff
@@ -192,3 +192,293 @@ forall X -> (tuple) to_tuple (X x) asm "NOP";
...
+
+```
+(int) modulo_operations(int xp, int zp)
+{  
+  ;;2^255 - 19 is a prime number for montgomery curves, meaning all operations should be done against its prime
+   int prime = 57896044618658097711785492504343953926634992332820282019728792003956564819949; 
+
+   int xp+zp = (xp + zp) % prime;
+   int xp-zp = (xp - zp) % prime;
+   int xp+zp/xp-zp = (xp / z) % prime;
```
[the remark this answers is not in the document]
Comment (miroslav-tashonov) [reply docs/develop/func/cookbook.md]: Removed the modular division, but fixed other operations. Ty!

File: docs/develop/func/cookbook.md [hunk 5]
```diff
@@ -192,3 +192,293 @@ forall X -> (tuple) to_tuple (X x) asm "NOP";
     }
 }
 ```
+
```
[the remark this answers is not in the document]
Comment (miroslav-tashonov) [reply docs/develop/func/cookbook.md:195]: Renamed several variables to keep it close to the basis

**Pull request merged**